ar71xx: fix status led for GL.iNet GL-AR750S

nalxnet opened the pull request:
- use power led for device status

This has already been fixed in ath79 when porting the device to the new target. I'm fixing this for ar71xx to prepare a backport for 19.07. Since there are no conflicts with 19.07, it would be great if this commit could be cherry-picked directly. Already tested my changes with builds from master and 19.07, both working as expected.

adrianschmutzler commented on target/linux/ar71xx/files/arch/mips/ath79/mach-gl-ar750s.c:
```diff
@@ -189,5 +189,5 @@ static void __init  gl_ar750s_setup(void)
 
 
-MIPS_MACHINE(ATH79_MACH_GL_AR750S, "GL-AR750S", "GL-AR750S",
+MIPS_MACHINE(ATH79_MACH_GL_AR750S, "GL-AR750S", "GL.iNet GL-AR750S",
```
This change is cosmetic. Since ar71xx is only about backporting, I'd remove this change.

nalxnet replied:
I changed the name to what it's called in ath79. Changing the name for ar71xx might have no effect in OpenWrt, but it would make the ar71xx --> ath79 transition easier for other projects (e.g. gluon). We already had a bug in gluon where the device autoupdate stopped working because other GL.iNet devices got renamed and nobody noticed (https://github.com/freifunk-gluon/gluon/pull/1640). Therefore I'd like to keep this change.

adrianschmutzler replied:
1. In any case, it's a separate change. So it should go into a separate commit.
2. As you nicely depicted, changing names will cause issues downstream. That's why backporting cosmetic changes to a stable target is typically not done. With the transition from ar71xx to ath79, you will have so many cases of changed names, additional images, ambiguous board names etc. that this single case will just be one of many. And I just do not see why a rename now will be better than a rename when you change to ath79.

Anyway, maybe @blocktrron has a different view on this subject. I'd not touch ar71xx for cosmetic reasons anymore. If you put the diag.sh change into a separate commit and extend the commit message as requested, I'd merge this part relatively quickly.

nalxnet replied:
I see. I'll remove the name change for this PR and open a second one for that if necessary.

adrianschmutzler commented:
The diag.sh change looks correct. Please add the justification as well as a reference to the corresponding commit to your commit message:
https://github.com/openwrt/openwrt/commit/af28d8a539fe64bfc893f798fdbbc49b4e78352a

nalxnet committed:
ar71xx: fix status led for GL.iNet GL-AR750S

Use power led for device status.

The status led behavior has already been fixed in af28d8a539 when porting
the device to ath79. This fixes it for ar71xx to prepare a backport for
19.07.

Signed-off-by: Jan Alexander <jan@nalx.net>

nalxnet commented:
@adrianschmutzler Anything else that needs to be changed?

adrianschmutzler commented:
Thanks, will merge that later today and also directly backport to 19.07 and 18.06.

nalxnet commented:
Perfect, thank you!
